terraform-beyond-layer0/example-2/main.tf

```diff
@@ -0,0 +1,91 @@
+#
+# ./example2/main.tf
+#
+
+provider "layer0" {
+  endpoint        = "${var.endpoint}"
+  token           = "${var.token}"
+  skip_ssl_verify = true
+}
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region     = "${var.region}"
+}
+
+# create one layer0 environment for each map value defined in variable 'environments'
+resource "layer0_environment" "demo" {
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
```

Comment:
Instead of a `map`, let's use a `list` for the `environments` variable. 

As an example, you can see how they use interpolation with the `public_subnets` variable here: https://github.com/terraform-community-modules/tf_aws_vpc/blob/master/main.tf#L65